File: jukebox/src/app/song/models/song.ts
export class Song {

  public title: string;
  public artist: string;
  public album: string;
  public id: number;

  constructor(id: number,title: string, artist?: string, album?: string)
  {
    this.id = id;
    this.title = title;
    this.artist = artist;
    this.album = album;
  }
}

File: Report/res/code/song.cs
public class Song{
    [Key]
    public int Id { get; set; }
    public string FilePath { get; set; }
    public string Title { get; set; }

    [ForeignKey(nameof(Album))]
    public int AlbumId { get; set; }
    public Album Album { get; set; }
    public DateTime LastTimeIndexed { get; set; }

    [NotMapped]
    public SongSource SourceType { get; set; } = SongSource.CustomBackend;
}
